### PR TITLE
add Strategy <> OperatorSet mapping in storage, and APIs and events

### PR DESCRIPTION
**Pull request by bowenli86**

### CODE REVIEW

File: src/contracts/core/AVSDirectory.sol [hunk 1]
```diff
@@ -451,6 +526,26 @@ contract AVSDirectory is
         }
     }
 
+    /**
+     * @notice Returns an array of strategies in the operatorSet.
+     * @param operatorSet The operatorSet to query.
+     * @param start The starting index of the array to query.
+     * @param length The amount of items of the array to return.
+     */
+    function getStrategiesInOperatorSet(
```
Comment (stevennevins) [inline src/contracts/core/AVSDirectory.sol:535]: @bowenli86 would it make your life easier if we just return all the enumerated strategies in the operator set?  I don't love this pattern for paginating these values with start and length params

File: src/contracts/core/AVSDirectoryStorage.sol [hunk 2]
```diff
@@ -69,6 +69,10 @@ abstract contract AVSDirectoryStorage is IAVSDirectory {
     /// @dev Each key is formatted as such: bytes32(abi.encodePacked(avs, uint96(operatorSetId)))
     mapping(bytes32 => EnumerableSet.AddressSet) internal _operatorSetMembers;
 
+    /// @notice Mapping: operatorSet => List of strategies that the operatorSet contains
+    /// @dev Each key is formatted as such: bytes32(abi.encodePacked(avs, uint96(operatorSetId)))
+    mapping(bytes32 => EnumerableSet.AddressSet) internal _operatorSetStrategies;
+
```
Comment (ypatil12) [inline src/contracts/core/AVSDirectoryStorage.sol:75]: Update storage gap

**Review by ypatil12 — COMMENTED** [review]
LGTM, just update gap

File: src/contracts/core/AVSDirectoryStorage.sol (hunk 2 above)
Comment (gpsanant) [reply src/contracts/core/AVSDirectoryStorage.sol:75]: done

**Review by ypatil12 — APPROVED** [review]

**Review by ypatil12 — APPROVED** [review]

**Pull request merged**